azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java

```diff
@@ -0,0 +1,135 @@
+   }
+
+   @Override
+   public ListenableFuture<Image> createImage(ImageTemplate template) {
+      final CloneImageTemplate cloneTemplate = (CloneImageTemplate) template;
+      final String id = cloneTemplate.getSourceNodeId();
+      final String storageAccountName = id.replaceAll("[^A-Za-z0-9 ]", "") + "stor";
+
+      // VM needs to be stopped before it can be generalized
+      String status = "";
+      api.getVirtualMachineApi(group).stop(id);
+      //Poll until resource is ready to be used
+      if (nodeSuspendedPredicate.apply(id)) {
+         return userExecutor.submit(new Callable<Image>() {
+            @Override
+            public Image call() throws Exception {
+               api.getVirtualMachineApi(group).generalize(id);
+
+               final String[] disks = new String[2];
+               URI uri = api.getVirtualMachineApi(group).capture(id, cloneTemplate.getName(), CONTAINER_NAME);
+               if (uri != null) {
+                  if (imageAvailablePredicate.apply(uri)) {
+                     List<ResourceDefinition> definitions = api.getJobApi().captureStatus(uri);
+                     if (definitions != null) {
+                        for (ResourceDefinition definition : definitions) {
+                           LinkedTreeMap<String, String> properties = (LinkedTreeMap<String, String>) definition.properties();
+                           Object storageObject = properties.get("storageProfile");
+                           LinkedTreeMap<String, String> properties2 = (LinkedTreeMap<String, String>) storageObject;
+                           Object osDiskObject = properties2.get("osDisk");
+                           LinkedTreeMap<String, String> osProperties = (LinkedTreeMap<String, String>) osDiskObject;
+                           Object dataDisksObject = properties2.get("dataDisks");
+                           ArrayList<Object> dataProperties = (ArrayList<Object>) dataDisksObject;
+                           LinkedTreeMap<String, String> datadiskObject = (LinkedTreeMap<String, String>) dataProperties.get(0);
+
+                           disks[0] = osProperties.get("name");
+                           disks[1] = datadiskObject.get("name");
+
+                           VirtualMachine vm = api.getVirtualMachineApi(group).get(id);
+                           String location = vm.location();
+                           final VMImage ref = VMImage.create(CUSTOM_IMAGE_PREFIX + group, CUSTOM_IMAGE_PREFIX + storageAccountName, disks[0], disks[1], location, false);
+                           return imageReferenceToImage.apply(ref);
+                        }
+                     }
+                  }
+               }
+               throw new UncheckedTimeoutException("Image was not created within the time limit: "
+                       + cloneTemplate.getName());
+            }
+         });
+      } else {
+         return null;
```

Comment:
The returned future should not be null. Better fail with an IllegalStateException here.